js/src/components/paid-ads/ads-campaign.js

```diff
@@ -25,19 +25,21 @@ import FaqsSection from './faqs-section';
 /**
  * Renders the container of the form content for campaign management.
  *
- * Please note that this component relies on an CampaignAssetsForm's context and custom adapter,
- * so it expects a `CampaignAssetsForm` to existing in its parents.
+ * Please note that this component relies on a CampaignAssetsForm's context and custom adapter,
+ * so it expects a `CampaignAssetsForm` to exist in its parents.
  *
  * @fires gla_documentation_link_click with `{ context: 'create-ads' | 'edit-ads' | 'setup-ads', link_id: 'see-what-ads-look-like', href: 'https://support.google.com/google-ads/answer/6275294' }`
  *
  * @param {Object} props React props.
  * @param {Campaign} [props.campaign] Campaign data to be edited. If not provided, this component will show campaign creation UI.
- * @param {() => void} props.onContinue Callback called once continue button is clicked.
+ * @param {Object} props.formProps Form props forwarded from `Form` component.
+ * @param {Function} props.formProps.isSubmitting Callback to display loading till Ads campaign is created.
+ * @param {Function} props.formProps.handleSubmit Callback to handle campaign creation.
  * @param {'create-ads'|'edit-ads'|'setup-ads'} props.trackingContext A context indicating which page this component is used on. This will be the value of `context` in the track event properties.
  */
 export default function AdsCampaign( {
 	campaign,
-	onContinue,
+	formProps: { isSubmitting, handleSubmit },
```

Comment:
@kt-12 Instead of passing an object, how about we keep the `onContinue` prop and add a new `isLoading` prop which will show the loading state?
It feels that `AdsCampaign` is like "tied" to the form.

```suggestion
	onContinue,
	isLoading,
```

Reply:
Makes sense. Fixed.